src/main/java/nextstep/subway/path/domain/AgeDiscount.java

```diff
@@ -0,0 +1,40 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.domain.Fare;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public enum AgeDiscount {
+
+    CHILD(age -> age >= 0 && age < 13, fare -> BigInteger.valueOf((long)((fare.longValue() - 350L) * 0.5))),
```

Comment:
요금을 계산하는 부분을 Fare의 책임으로 이전해볼 수도 있을 것 같아요 :)

```suggestion
    CHILD(age -> age >= 0 && age < 13, fare -> fare.minus(DEFAULT_DISCOUNT_PRICE).multiply(CHIILD_DISCOUNT_RATE),
```


추가적으로 임의의 숫자를 사용하는 매직넘버는소스 코드를 읽기 어렵게 만드는데요 !

이 부분을 어떻게 개선할 수 있을까요? 관련 링크 남겨놓겠습니다 :)

[https://www.slipp.net/questions/356](https://www.slipp.net/questions/356)

Reply:
네넵